### PR TITLE
Change setup

### PR DESCRIPTION
**Pull request by bitPogo**
## Description

This adjusts the setup of the SDK to the latest version.
Also it changes the git workflows to the cloud runners.

## How is it being implemented?

It rolls out the dependency plugin for the SDK and updates the D4L dependencies. This also contains adjustments of the DataRestrictionException.
Also the workflows got adjusted to use the cloud runners. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the changelog accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -15,22 +15,42 @@ ifndef::env-github[]
...
+        <h3>Data4Life FHIR Helper SDK</h3>
+    </p>
+    <p><!-- PROJECT DESCRIPTION -->
+        Provides builder and extension to create and interact with FHIR objects
+    </p>
+    <p><!-- PROJECT DOCUMENTATION -->
+        <a href="README.adoc"><strong>Explore the docs »</strong></a>
+    </p>
+    <p><!-- PROJECT ISSUES/FEATURES -->
+        <a href="https://github.com/d4l-data4life/hc-helpers-sdk-kmp/issues">Report Bug</a>
```
Comment (wmontwe) [inline README.adoc:34]: the link is wrong -> fhir missing

File: README.adoc [hunk 2]
```diff
@@ -15,22 +15,42 @@ ifndef::env-github[]
...
+        ·
+        <a href="https://github.com/d4l-data4life/hc-helpers-sdk-kmp/issues">Request Feature</a>
```
Comment (wmontwe) [inline README.adoc:36]: link

File: README.adoc [hunk 3]
```diff
@@ -15,22 +15,42 @@ ifndef::env-github[]
...
+    </p>
+    <p><!-- PROJECT BADGES see badges.adoc how to change them -->
+        <a href="https://github.com/d4l-data4life/hc-helpers-sdk-kmp/releases">
```
Comment (wmontwe) [inline README.adoc:39]: link

File: README.adoc [hunk 4]
```diff
@@ -15,22 +15,42 @@ ifndef::env-github[]
...
+            <img src="assets/images/badge-release-latest.svg" alt="Latest version"/>
+        </a>
+        <a href="https://github.com/d4l-data4life/hc-helpers-sdk-kmp/actions">
```
Comment (wmontwe) [inline README.adoc:42]: link

File: README.adoc [hunk 5]
```diff
@@ -15,22 +15,42 @@ ifndef::env-github[]
...
+            <img src="https://github.com/d4l-data4life/hc-helpers-sdk-kmp/actions/workflows/d4l-ci-latest-version.yml/badge.svg" alt="Build Status"/>
```
Comment (wmontwe) [inline README.adoc:43]: link

**Review by wmontwe — CHANGES_REQUESTED** [review]
just some wrong links

File: README.adoc (hunk 1 above)
Comment (bitPogo) [reply README.adoc]: done

File: README.adoc (hunk 2 above)
Comment (bitPogo) [reply README.adoc]: done

File: README.adoc (hunk 3 above)
Comment (bitPogo) [reply README.adoc]: done

File: README.adoc (hunk 4 above)
Comment (bitPogo) [reply README.adoc]: done

File: README.adoc (hunk 5 above)
Comment (bitPogo) [reply README.adoc]: done

**Review by wmontwe — APPROVED** [review]
:+1:

**Pull request merged**